src/states/elder/mod.rs

```diff
@@ -714,24 +714,35 @@ impl Elder {
             msg.routing_message()
         );
 
-        let dst = msg.message_dst();
-        let signed_msg = msg.signed_routing_message();
-        if self.in_location(dst) {
-            self.check_signed_message_trust(signed_msg)?;
+        if self.in_location(msg.message_dst()) {
+            let signed_msg = msg.signed_routing_message();
+            let handle = self.check_signed_message_trust(signed_msg)?;
             self.check_signed_message_integrity(signed_msg)?;
-            self.update_our_knowledge(signed_msg);
 
-            if dst.is_multiple() {
-                // Broadcast to the rest of the section.
-                if let Err(error) = self.send_signed_message(&msg) {
-                    debug!("{} Failed to send {:?}: {:?}", self, signed_msg, error);
+            if handle {
+                self.update_our_knowledge(signed_msg);
+                self.routing_msg_queue.push_back(signed_msg.clone());
+            }
+
+            if !msg.message_dst().is_multiple() {
+                if !handle {
+                    // Message is only for us but we can't handle it yet. Backlog it.
```

Comment:
Why should it be if it is only for us?

Reply:
This is to cover the case where we receive a message with dst = Node whose proof is too new. But I guess this cannot really happen, because we use the `members_knowledge` thing for messages sent to nodes. So probably best to treat this case as normal untrusted message and fail. I'll change it.